src/earn/prepareTransactions.ts

```diff
@@ -79,62 +78,52 @@ export function usePrepareSupplyTransactions() {
 }
 
 export async function prepareWithdrawAndClaimTransactions({
-  amount,
-  token,
+  pool,
   walletAddress,
   feeCurrencies,
-  rewardsTokens,
-  poolTokenAddress,
-}: {
-  amount: string
-  token: TokenBalance
-  poolTokenAddress: Address
-  walletAddress: Address
-  feeCurrencies: TokenBalance[]
-  rewardsTokens: Token[]
-}) {
-  const baseTransactions: TransactionRequest[] = []
-
-  if (!token.address || !isAddress(token.address)) {
-    // should never happen
-    throw new Error(`Cannot use a token without address. Token id: ${token.tokenId}`)
-  }
-
-  const amountToWithdraw = maxUint256 // Withdraws entire balance https://docs.aave.com/developers/core-contracts/pool#withdraw
-
-  baseTransactions.push({
-    from: walletAddress,
-    to: networkConfig.arbAavePoolV3ContractAddress,
-    data: encodeFunctionData({
-      abi: aavePool,
-      functionName: 'withdraw',
-      args: [token.address, amountToWithdraw, walletAddress],
-    }),
-  })
-
-  rewardsTokens.forEach(({ balance, decimals, address }) => {
-    const amountToClaim = parseUnits(balance, decimals)
-
-    if (!isAddress(address)) {
-      // should never happen
-      throw new Error(`Cannot use a token without address. Token id: ${token.tokenId}`)
-    }
-
-    baseTransactions.push({
-      from: walletAddress,
-      to: networkConfig.arbAaveIncentivesV3ContractAddress,
-      data: encodeFunctionData({
-        abi: aaveIncentivesV3Abi,
-        functionName: 'claimRewardsToSelf',
-        args: [[poolTokenAddress], amountToClaim, address],
-      }),
+  hooksApiUrl,
+  positionsWithClaimableRewards,
+}: PrepareWithdrawAndClaimParams) {
+  const { transactions: withdrawTransactions }: { transactions: RawShortcutTransaction[] } =
+    await triggerShortcutRequest(hooksApiUrl, {
+      address: walletAddress,
+      appId: pool.appId,
+      networkId: pool.networkId,
+      shortcutId: 'withdraw',
+      tokens: [
+        {
+          tokenId: pool.dataProps.withdrawTokenId,
+          amount: pool.balance,
+          useMax: true,
+        },
+      ],
+      ...pool.shortcutTriggerArgs?.withdraw,
+    })
+  const claimTransactions = await Promise.all(
+    positionsWithClaimableRewards.map(async (position): Promise<RawShortcutTransaction[]> => {
+      const { transactions }: { transactions: RawShortcutTransaction[] } =
+        await triggerShortcutRequest(hooksApiUrl, {
+          address: walletAddress,
+          appId: pool.appId,
+          networkId: pool.networkId,
+          shortcutId: 'claim-rewards',
+          ...position.shortcutTriggerArgs?.['claim-rewards'],
```

Comment:
`positionsWithClaimableRewards` selector doesn't guarantee that the position has a shortcut id named `claim-rewards` right? and I've see different positions using different shortcutId's for claims, so with this effectively we're enforcing any earn position / pool with claimable rewards have this as the id? I wonder if this has implications with the Dapp shortcuts claim flow, where we might expect this id to be unique.

Reply:
Hmm, with this and the the other feedback about `ClaimPosition` in the tests i realized that there is actually not a strong reason for this to be a ClaimPosition instead of just a Position especially when we have `rewardsPositionIds`.